exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/IncubatingUtil.java

```diff
@@ -28,11 +28,28 @@
  */
 public class IncubatingUtil {
 
+  private static final boolean INCUBATOR_AVAILABLE;
+
+  static {
+    boolean incubatorAvailable = false;
+    try {
+      Class.forName("io.opentelemetry.api.incubator.common.ExtendedAttributes");
+      incubatorAvailable = true;
+    } catch (ClassNotFoundException e) {
+      // Not available
+    }
+    INCUBATOR_AVAILABLE = incubatorAvailable;
+  }
+
   private static final byte[] EMPTY_BYTES = new byte[0];
   private static final KeyValueMarshaler[] EMPTY_REPEATED = new KeyValueMarshaler[0];
 
   private IncubatingUtil() {}
 
+  public static boolean isExtendedLogRecordData(LogRecordData logRecordData) {
+    return INCUBATOR_AVAILABLE && logRecordData instanceof ExtendedLogRecordData;
```

Comment:
I'm nervous about this. 

Recall that the test suites are set up that the main test suite intentionally doesn't include the incubator, to ensure that everything works without the incubator present. 

The build was working locally when this was equal to `return logRecordData instanceof ExtendedLogRecordData`.. which suggests that `instanceOf` doesn't try to load classes, since if it did, it would load `ExtendedLogRecordData` and generate an exception.

I included this `INCUBATOR_AVAILABLE` check first in the hopes that if the incubator is not present ,the `instanceOf` check will never be reached and we'll never load the class. But not feeling confident about what the JVM will do here.

Reply:
maybe it's worth storing the `ExtendedLogRecordData` class object in a static final and using `isAssignableFrom()` to avoid the worry? I wouldn't expect perf to be much different

Reply:
I think maybe a comment for our future selves that says something like "this is ok, because the instanceof is only evaluated when the incubator is present, so the class _can_ be found" or something, so we remember.